uk.co.compucorp.civicrm.tasksassignments/api/v3/Document.php

```diff
@@ -241,18 +242,30 @@ function _civicrm_api3_document_create_spec(&$params) {
       'api.default' => 'user_contact_id',
   ];
 
-  // replace custom data options with their column name
-  foreach ($params as $key => $param) {
+  $metadata = civicrm_api3('CustomField', 'get', [
+    'custom_group_id' => "Activity_Custom_Fields",
+    'options' => ['limit' => 0],
+  ])['values'];
 
-    $parent = CRM_Utils_Array::value('extends', $param);
+  foreach ($params as $key => $param) {
+    $parent = ArrayHelper::value('extends', $param);
     $isCustom = substr($key, 0, 7) === 'custom_';
 
     if ($isCustom && $parent === 'Activity' && isset($param['column_name'])) {
+      $customFieldId = substr($key, strrpos($key, '_') + 1);
```

Comment:
Couldn't understand what was happening here at a glance. Maybe a comment like:
//For example custom_85 has an ID of 85 as a custom field.
What do you think?